### PR TITLE
Updates and cleanups

### PR DESCRIPTION
**Pull request by jtt**
* Update to Rust edition 2021
* Update deps
* Update `clap` to 3.1 series.
* Fix clippy lints

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,16 +2,16 @@
 name = "pscan"
 version = "0.1.0"
```
Comment (atonkyra) [inline Cargo.toml:3]: Would consider a version bump to 0.1.1

File: src/config.rs [hunk 2]
```diff
@@ -23,38 +23,38 @@ pub const ARG_VERBOSE: &str = "verbose";
 pub const PSCAN_VERSION: &str = "0.1.0";
```
Comment (atonkyra) [inline src/config.rs:23]: Would consider a version bump to 0.1.1

**Review by atonkyra — COMMENTED** [review]
I'd bump the version. Otherwise LGTM.

File: src/config.rs (hunk 2 above)
Comment (jtt) [reply src/config.rs]: Bumped to 0.2.0-dev

**Review by atonkyra — APPROVED** [review]

**Pull request merged**